dynamicio/handlers/file.py

```diff
@@ -0,0 +1,83 @@
+"""File handlers for dynamicio."""
+from copy import deepcopy
+from pathlib import Path
+from threading import Lock
+from typing import Any, Dict
+
+import pandas as pd
+from pydantic import Field
+
+from dynamicio import utils
+from dynamicio.base import BaseResource
+from dynamicio.inject import check_injections, inject
+
+hdf_lock = Lock()
+
+
+class BaseFileResource(BaseResource):
```

Comment:
`class KafkaResource(BaseFileResource):` will also be crucial for early adoption for some repos

Reply:
wdym by `class KafkaResource(BaseFileResource):` The Kafka one only writes and is separate from any file logic I think/hope?

Reply:
Oh I see. No, I meant `class KafkaResource(BaseResource)`. Regarding S3, actually `read_csv` and `read_parquet` should already be able to read from S3 if you pass in a `S3://` string. So that functionality is already there?

Reply:
Well, yes and no. The old version did some funcy getattr to actually use the file implementation under the hood. But with one extra functionality on top, downloading the file to a temporary location first, unless specifying that there is no disk space. That is faster to do apparently.

I can put that part into this PR (it's already done) - or into a later PR.